### ISSUE TITLE
Can't create 4K video at 60fps (RenderToFileAsync)

### ISSUE THREAD

**Issue by jtorjo** [microsoft/WindowsAppSDK#832]
**Describe the bug**

**Steps to reproduce the bug**

    var clip = MediaClip.CreateFromColor(Colors.Transparent, TimeSpan.FromMilliseconds(1000));
    var composition = new MediaComposition();
    composition.Clips.Add(clip);
    var  encoding_profile =  MediaEncodingProfile.CreateMp4( VideoEncodingQuality.Uhd2160p);
    encoding_profile.Video.Width = 3840;
    encoding_profile.Video.Height = 2160;
    encoding_profile.Video.FrameRate.Numerator = 60;
    encoding_profile.Video.FrameRate.Denominator = 1;

    var mp4_name = ApplicationData.Current.RoamingFolder.Path + "\\whatever.mp4";
    var folder = await StorageFolder.GetFolderFromPathAsync(Path.GetDirectoryName(mp4_name));
    var file = await folder.CreateFileAsync(Path.GetFileName(mp4_name), CreationCollisionOption.GenerateUniqueName);
    await composition.RenderToFileAsync(file, MediaTrimmingPreference.Precise, encoding_profile);

This code throws: `{"A valid type has not been set for this stream or a stream that it depends on. (Exception from HRESULT: 0xC00D6D60)"}`

So basically, I can't ever create a 4k/60fps video using Windows API.

I really really really hope I'm doing something wrong.

**Expected behavior**

The above code should work.

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| May 2020 Update (19041)            | Yes  |

**Comment by Jaiganeshkumaran** [comment]
The built-in media APIs doesn't support 4K I guess because even the Photos app limits the target to 1080p and it uses the same APIs.

**Comment by jtorjo** [comment]
@Jaiganeshkumaran Please stop trying to be helpful when you know nothing of the issue in question. 
4K is supported at 30fps.

**Comment by Jaiganeshkumaran** [comment]
> @Jaiganeshkumaran Please stop trying to be helpful when you know nothing of the issue in question.
> 4K is supported at 30fps.

Then why doesn't the Photos app support 4K in export options? It also uses the same MediaClip APIs.

**Comment by jtorjo** [comment]
> Then why doesn't the Photos app support 4K in export options? It also uses the same MediaClip APIs.

Why would I care? I do care however, about my original issue. 
So please Microsoft, answer :)

**Comment by riverar** [comment]
Can repro across multiple machines / OS versions. Attached repro.

[ReunionIssue832Repro.zip](https://github.com/microsoft/ProjectReunion/files/6503027/ReunionIssue832Repro.zip)

**Comment by riverar** [comment]
I can also repro with generally anything that would be classified as H.264/AVC Level 5.2 or above (e.g. 1920x1080@172). Seems to be a MediaComposition limitation but I can't nail down any relevant documentation.

**Comment by jonwis** [comment]
@aklemets or @sudhakar-vp-zz can you help route this one?

**Comment by mattrwoz** [comment]
I tried the exact sample code you provided and was not able to reproduce any failure.  It successfully made a 4k H264 video file in the roaming data directory.

Would you be able to collect a trace so we could identify why this is not working on your system?

Trace instructions:
1.	Download mftracelog.exe https://github.com/microsoft/media-foundation/releases/download/V1/mftracelog.zip
2.	Launch mfracelog.exe
3.	Select "Full trace" option and click "Start Tracing"
4.	Repro the scenario 
5.	Click "Stop Tracing" on mftracelog.exe. It will open up explorer with mf.etl, version, dxdiag.exe files. Zip the files and share.

**Comment by riverar** [comment]
@mattrwoz Curious, because this repros in a clean VM on 21382 co_release. Sounds like it might be fixed on whatever dogfood build of Windows 10 you're on? 😃

Requested traces (from VM, so no PII issue):
[mftraces_21382.co_release.zip](https://github.com/microsoft/ProjectReunion/files/6513041/mftraces_21382.co_release.zip)

**Comment by jtorjo** [comment]
Hi guys, any ETA on this one?
It is pretty major IMHO.

**Comment by aklemets** [comment]
When you call `MediaEncodingProfile.CreateMp4(VideoEncodingQuality.Uhd2160p)`, the media stack assumes that the video will be encoded using Level 5.1. At the point of you making the call, the API does not yet know the exact resolution and frame rate that you intend to use.

At 60 fps the correct H.264 encoding level is probably 5.2. Whether the H.264 encoder accepts the wrong level being specified depends on whether the stack uses the software encoder or a 3rd party encoder running on the GPU. For hardware-based encoders you may get different results depending on the vendor and model of your video card.
That is why the bug does not happen for everyone.

To work around the bug you could try to explicitly set the H.264 encoding level to 5.2, like this:

`encoding_profile.Video.Properties["{96F66574-11C5-4015-8666-BFF516436DA7}"] = 52;`

**Comment by jtorjo** [comment]
Thanks @aklemets !
Yes, just tried it and it works - I also took a look at https://en.wikipedia.org/wiki/Advanced_Video_Coding#Levels -- so I can adjust this to my needs.
It is a bit sad that this does not happen by default, I'm pretty sure a lot of people would think `RenderToFile` just doesn't work correctly.
Anyway, glad it works!

One more thing - since you're definitely an expert on this: do you know off the top of your head the solution to this (if any): https://github.com/microsoft/WindowsAppSDK/discussions/1301 
Thanks!
